File: java/Bink/src/net/qiujuer/blink/SendDelivery.java
package net.qiujuer.blink;

/**
 * Send delivery interface
 */
public interface SendDelivery {
    /**
     * Parses a start response from the sender.
     */
    public void postSendStart(SendEntity entity);

    /**
     * Parses a end response from the sender.
     */
    public void postSendEnd(SendEntity entity, boolean isSuccess);

    /**
     * Parses a progress response from the sender.
     */
    public void postSendProgress(SendEntity entity, int total, int cur);
}

File: C Sharp/Blink/Blink/Listener/Delivery/SendDelivery.cs
using Net.Qiujuer.Blink.Core;
using System;
namespace Net.Qiujuer.Blink.Listener.Delivery
{
    /// <summary>
    /// Send delivery interface
    /// </summary>
    public interface SendDelivery : IDisposable
    {
        /**
         * Parses a progress response from the sender.
         */
        void PostSendProgress(SendPacket entity, float progress);
    }
}
